components/services/bat_ledger/bat_ledger_impl.cc

```diff
@@ -335,4 +335,22 @@ void BatLedgerImpl::HasSufficientBalanceToReconcile(
   std::move(callback).Run(ledger_->HasSufficientBalanceToReconcile());
 }
 
+// static
+void BatLedgerImpl::OnAddressesForPaymentId(
+    CallbackHolder<GetAddressesForPaymentIdCallback>* holder,
+    std::map<std::string, std::string> addresses) {
+  if (holder->is_valid())
+    std::move(holder->get()).Run(mojo::MapToFlatMap(addresses));
+  delete holder;
+}
+
+void BatLedgerImpl::GetAddressesForPaymentId(
+    GetAddressesForPaymentIdCallback callback) {
+  // delete in OnAddressesForPaymentId
+  auto* holder = new CallbackHolder<GetAddressesForPaymentIdCallback>(
```

Comment:
we always use pointers here, so I just used pattern that was established by @yrliou

Reply:
We call move on that raw pointer here, which has no effect. If the intent is to change ownership, we should convert to `unique_ptr`

Reply:
not sure why it was done like this, cc @yrliou

Reply:
We'll move to separate issue if this is being used other than here.

Reply:
This is fine.

I was misreading the `std::move` as `std::move(holder)` rather than `std::move(holder->get())` which is moving the callback template rather than the raw pointer.